xarray/backends/api.py

```diff
@@ -477,6 +485,7 @@ def maybe_decode_store(store, lock=False):
                 chunks,
                 drop_variables,
                 use_cftime,
+                decode_timedelta,
```

Comment:
In theory, it would be better organized to add this argument after `decode_times`, but for now putting it at the end is a good choice to preserve backwards compatibility. I made a note about making most `open_dataset` arguments keyword only in https://github.com/pydata/xarray/issues/4080.